Libraries/LibWeb/CSS/StyleValues/ShorthandStyleValue.cpp

```diff
@@ -396,6 +396,17 @@ String ShorthandStyleValue::to_string(SerializationMode mode) const
         return builder.to_string_without_validation();
     }
     default:
+        auto all_properties_same_value = true;
+        auto first_property_value = m_properties.values.first();
+        for (auto i = 1u; i < m_properties.values.size(); ++i) {
+            if (m_properties.values[i] != first_property_value) {
+                all_properties_same_value = false;
+                break;
+            }
+        }
+        if (all_properties_same_value)
+            return first_property_value->to_string(mode);
```

Comment:
We have pretty similar logic for keywords at the top of this function; seems easier to make it work there?

Reply:
Doing the check here means that we don't do it for properties that have been special-cased.

Let me check if doing the check sooner leads to any incorrect behavior.

Reply:
Ah right, that makes sense. Seems like a lot of specialized cases have similar deduplicating logic as well, would be interesting to see the results.

Reply:
Doing the check sooner leads to regressions for a few special cases, so I don't think it's safe to move.